pkg/controller/deployment/util/deployment_util.go

```diff
@@ -51,21 +51,169 @@ const (
+		Reason:             reason,
+		Message:            message,
+	}
+}
+
+// GetCondition returns the latest deployment condition it will find with the provided type
+// and status.
+func GetCondition(status extensions.DeploymentStatus, cType extensions.DeploymentConditionType, cStatuses ...api.ConditionStatus) *extensions.DeploymentCondition {
+	statuses := map[api.ConditionStatus]struct{}{}
+	for _, cStatus := range cStatuses {
+		statuses[cStatus] = struct{}{}
+	}
+
+	conditions := []extensions.DeploymentCondition{}
+	for _, c := range status.Conditions {
+		_, ok := statuses[c.Status]
+		if c.Type == cType && ok {
+			conditions = append(conditions, c)
+		}
+	}
+
+	if len(conditions) == 0 {
+		return nil
+	}
+
+	sort.Sort(ByLastTransitionTime(conditions))
+
+	return &conditions[0]
+}
+
+// SetCondition updates the deployment to include the provided condition. If the condition
+// already exists and overwriteIfExists is true, then the condition will be replaced. True
+// will be returned if there is an update in the deployment conditions.
+func SetCondition(status extensions.DeploymentStatus, condition extensions.DeploymentCondition, overwriteIfExists bool) (extensions.DeploymentStatus, bool) {
+	if overwriteIfExists {
+		newConditions, _ := filterOutCondition(status, condition.Type, condition.Status)
+		status.Conditions = append(newConditions, condition)
+		return status, true
+	}
+
+	if cond := GetCondition(status, condition.Type, condition.Status); cond == nil {
+		status.Conditions = append(status.Conditions, condition)
+		return status, true
+	}
+
+	return status, false
+}
+
+// RemoveCondition removes the deployment condition with the provided type and status from
+// the deployment conditions. It returns true if there was a removal, false otherwise.
+func RemoveCondition(status extensions.DeploymentStatus, cType extensions.DeploymentConditionType, cStatus api.ConditionStatus) (extensions.DeploymentStatus, bool) {
+	conditions, found := filterOutCondition(status, cType, cStatus)
+	status.Conditions = conditions
+	return status, found
+}
+
+// filterOutCondition returns a new slice of deployment conditions without conditions with the
+// provided type and status.
+func filterOutCondition(status extensions.DeploymentStatus, cType extensions.DeploymentConditionType, cStatus api.ConditionStatus) ([]extensions.DeploymentCondition, bool) {
+	newConditions := []extensions.DeploymentCondition{}
+	found := false
+
+	for _, c := range status.Conditions {
+		if c.Type == cType && c.Status == cStatus {
+			found = true
+			continue
+		}
+		newConditions = append(newConditions, c)
+	}
+
+	return newConditions, found
+}
+
+// ConditionExists returns if the provided condition already exists in the deployment conditions and the
+// only difference is in the timestamp.
+func ConditionExists(status extensions.DeploymentStatus, condition extensions.DeploymentCondition) bool {
+	for _, c := range status.Conditions {
+		if c.Type == condition.Type &&
```

Comment:
@kargakis might be smart if we ever add a new field into status, this will not be yet another place to not forget.